idea/idea-fir/src/org/jetbrains/kotlin/idea/quickfix/fixes/WrapWithSafeLetCall.kt

```diff
@@ -0,0 +1,290 @@
+            // In addition, if there is no parent that is at a nullable position, then we don't offer wrapping with let either because
+            // it still doesn't fix the code. Hence, the plain safe call operator is a better fix.
+            findParentExpressionAtNullablePosition(nullableExpression) == null
+        ) {
+            return emptyList()
+        }
+        return createWrapWithSafeLetCallInputForNullableExpression(nullableExpression, isImplicitInvokeCallToMemberProperty)
+    }
+
+    private fun KtAnalysisSession.createWrapWithSafeLetCallInputForNullableExpression(
+        nullableExpression: KtExpression?,
+        isImplicitInvokeCallToMemberProperty: Boolean = false,
+        parentExpression: KtExpression? = findParentExpressionAtNullablePosition(nullableExpression)
+            ?: nullableExpression?.typableParentExpression
+    ): List<IntentionAction> {
+        if (nullableExpression == null || parentExpression == null) return emptyList()
+        val existingNames = nullableExpression.containingKtFile.getScopeContextForPosition(nullableExpression).scopes.getCallableNames()
+            .mapNotNull { it.identifierOrNullIfSpecial }
+        // Note, the order of the candidate matters. We would prefer the default `it` so the generated code won't need to declare the
+        // variable explicitly.
+        val candidateNames = listOfNotNull("it", getDeclaredParameterNameForArgument(nullableExpression))
+        val suggestedName = FirKotlinNameSuggester.suggestNameByMultipleNames(candidateNames) { it !in existingNames }
+        return listOf(
+            HLQuickFix(
+                parentExpression,
+                Input(nullableExpression.createSmartPointer(), suggestedName, isImplicitInvokeCallToMemberProperty),
+                applicator
+            )
+        )
+    }
+
+    private fun KtAnalysisSession.getDeclaredParameterNameForArgument(argumentExpression: KtExpression): String? {
+        val valueArgument = argumentExpression.parent as? KtValueArgument ?: return null
+        val successCallTarget =
+            (argumentExpression.parentOfType<KtCallExpression>()?.resolveCall()?.targetFunction?.candidates?.singleOrNull()) ?: return null
+        return successCallTarget.valueParameters.getOrNull(valueArgument.argumentIndex)?.name?.identifierOrNullIfSpecial
+    }
+
+    private fun KtAnalysisSession.findParentExpressionAtNullablePosition(expression: KtExpression?): KtExpression? {
+        if (expression == null) return null
+        var current = expression.typableParentExpression
+        while (current != null && !isExpressionAtNullablePosition(current)) {
+            current = current.typableParentExpression
+        }
+        return current
+    }
+
+    private fun KtAnalysisSession.isExpressionAtNullablePosition(expression: KtExpression): Boolean {
+        val parent = expression.parent
+        return when {
+            parent is KtProperty && expression == parent.initializer -> {
```

Comment:
Can you add a test for when property is not marked nullable? I think something like `changedCallTypeMismatch.kt` but with `val res: Int = ...`

Reply:
Sounds good! Let me do that!